### PR TITLE
Add WAF section. Add IPv6 in VPC, EC2, CloudFront, WAF, CLB, and ALB sections.

### PR DESCRIPTION
**Pull request by ClydeMachine**
With the introduction of IPv6 in VPC, it is important to note 1) that it is now available after having been asked for by users for literally _years_ now, and 2) that there are some caveats we'll need to get used to, in particular the segregated entries in Security Groups and Route Tables for the separate protocols.

Update: I went ahead and took care of the other services that utilize IPv6, as well as instantiate the WAF section. Those commits are below.

### CODE REVIEW

**Comment by ClydeMachine** [comment]
With these latest commits, pull request #347 completes issue #160 for covering all services currently supporting IPv6.

**Comment by ThanosBaskous** [comment]
Thanks for taking this on, @ClydeMachine! These changes look good to me. I have a few micro-nits (not included) around use of curly quotes but can take care of those on my own after merge.

Over to @jlevy for a second review.

File: README.md [hunk 1]
```diff
@@ -1413,7 +1422,7 @@ CloudFront
 
 ### CloudFront Tips
 
--	🐥**IPv6** is [now supported](https://aws.amazon.com/about-aws/whats-new/2016/10/ipv6-support-for-cloudfront-waf-and-s3-transfer-acceleration/)!
+-	🐥**IPv6** is [supported](https://aws.amazon.com/about-aws/whats-new/2016/10/ipv6-support-for-cloudfront-waf-and-s3-transfer-acceleration/). This is a configurable setting, and is enabled by default on new CloudFront distribution. IPv6 support extends to the use of WAF with CloudFront.
```
Comment (jlevy) [inline README.md]: Post merge: distribution -> distributions

File: README.md [hunk 2]
```diff
@@ -1363,6 +1367,9 @@ VPCs, Network Security, and Security Groups
 -   **VPC Flow Logs** allow you to monitor the network traffic to, from, and within your VPC. Logs are stored in CloudWatch Logs groups, and can be used for security monitoring (with third party tools), performance evaluation, and forensic investigation.
 	-	See the [VPC Flow Logs User Guide](http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/flow-logs.html) for basic information.
 	-	See the [flowlogs-reader](https://github.com/obsrvbl/flowlogs-reader) CLI tool and Python library to retrieve and work with VPC Flow Logs.
+- **IPv6** [is available in VPC.](https://aws.amazon.com/blogs/aws/new-ipv6-support-for-ec2-instances-in-virtual-private-clouds/) Along with this announcement came the introduction of the Egress-Only Internet Gateway. In cases where one would use NAT Gateways to enable egress-only traffic for their VPC in IPv4, one can use an Egress-Only Internet Gateway for the same purpose in IPv6.
```
Comment (jlevy) [inline README.md]: Post merge: Link on Egress-Only Internet Gateway

File: README.md [hunk 3]
```diff
@@ -1026,6 +1029,7 @@ ALB
 -	Instances in the ALB’s target groups have to either have a single, fixed healthcheck port (“EC2 instance”-level healthcheck) or the healthcheck port for a target has to be the same as its application port (“Application instance”-level healthcheck) - you can't configure a per-target healthcheck port that is different than the application port.
 -	ALBs are VPC-only (they are not available in EC2 Classic)
 -	In a target group, if there is no healthy target, all requests are routed to all targets. For example, if you point a listener at a target group containing a single service that has a long initialization phase (during which the health checks would fail), requests will reach the service while it is still starting up.
+- Although it has been announced as coming soon, [ALBs do not presently support IPv6.](https://aws.amazon.com/about-aws/whats-new/2016/12/announcing-internet-protocol-version-6-support-for-ec2-instances-in-amazon-virtual-private-cloud/) Even if the ALB is launched in an IPv6-enabled VPC and subnet, it will only have IPv4 addressing.
```
Comment (jlevy) [inline README.md]: Post merge: Add period.

Comment (jlevy) [inline README.md]: Post merge: "as of December 2016,"

**Comment by jlevy** [comment]
👍 Looks good. Marked a few copy editing nits we can take care of post merge.

**Pull request merged**